common/util/file_util.h

```diff
@@ -34,6 +34,12 @@ absl::string_view Basename(absl::string_view filename);
 // there is no "." in the basename, this is equivalent to file::Basename(path).
 absl::string_view Stem(absl::string_view filename);
 
+// Search for 'filename' file starting at 'start' (can be file or directory)
+// and going upwards the directory tree. Returns true if anything is found and
+// puts the found file in 'result'.
+bool UpwardFileSearch(absl::string_view start, absl::string_view filename,
+                      std::string& result);
```

Comment:
[style] pass modified objects by pointers instead of reference, so `std::string* result`.